include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -410,11 +413,19 @@ class transform_iterator
     typedef typename ::std::iterator_traits<_Iter>::pointer pointer;
     typedef typename ::std::iterator_traits<_Iter>::iterator_category iterator_category;
 
-    transform_iterator(_Iter __it = _Iter(), _UnaryFunc __unary_func = _UnaryFunc())
-        : __my_it_(__it), __my_unary_func_(__unary_func)
+    //default constructor will only be present if both the unary functor and iterator are default constructible
+    transform_iterator() = default;
+
+    //only enable this constructor if the unary functor is default constructible
+    template <typename _UnaryFuncLocal = _UnaryFunc,
+              ::std::enable_if_t<::std::is_default_constructible_v<_UnaryFuncLocal>, int> = 0>
+    transform_iterator(_Iter __it) : __my_it_(__it), __my_unary_func_()
```

Comment:
Technically, we can do `std::move`. And you don't actually need `__my_unary_func_()`

```suggestion
    transform_iterator(_Iter __it) : __my_it_(::std::move(__it))
```

Reply:
Added, but see my note about this possibly being a breaking change:  
https://github.com/oneapi-src/oneDPL/pull/1445/files#r1535775332